File: backend/app/schemas/food_analysis.py
"""음식 사진 분석 API 스키마 - contracts/food.contract.ts와 동기화"""

from typing import Literal

from pydantic import BaseModel, Field

from app.schemas.food import FoodItem


class DetectedFood(BaseModel):
    detected_name: str = Field(description="AI가 인식한 음식 이름")
    confidence: Literal["high", "medium", "low"]
    matched_foods: list[FoodItem] = Field(default_factory=list)


class FoodAnalysisMeta(BaseModel):
    total_detected: int
    model: str = "gpt-4o"


class FoodAnalysisResponse(BaseModel):
    data: list[DetectedFood]
    meta: FoodAnalysisMeta

File: frontend/src/mocks/handlers/foodAnalysis.ts
import { http, HttpResponse } from 'msw';
import { mockFoods } from '../data/food';
import type { FoodAnalysisResponse } from '../../../contracts/food.contract';
import { API_BASE } from '../../config';

export const foodAnalysisHandlers = [
  http.post(`${API_BASE}/api/v1/foods/analyze-image`, async () => {
    // 목 응답: 김치찌개, 비빔밥을 인식한 것처럼 반환
    const response: FoodAnalysisResponse = {
      data: [
        {
          detected_name: '김치찌개',
          confidence: 'high',
          matched_foods: mockFoods.filter((f) => f.name.includes('김치찌개')),
        },
        {
          detected_name: '비빔밥',
          confidence: 'medium',
          matched_foods: mockFoods.filter((f) => f.name.includes('비빔밥')),
        },
      ],
      meta: { total_detected: 2, model: 'gemini-2.0-flash' },
    };

    return HttpResponse.json(response);
  }),
];
